pkg/mcp/server/server.go

```diff
@@ -465,7 +465,13 @@ func (con *connection) receive() {
 			con.reqError = err
 			return
 		}
-		con.requestC <- req
+		select {
```

Comment:
Doesn't stream.Recv() return an error when the stream context is canceled?

Reply:
Yes, it does. But the issue is  after receive returns, but when send to `requestC ` blocked.

Reply:
We return an error and skip posting to the `requestC` on receive error. I'm not sure how re-checking the context below would completely solve the problem, since this still races with posting to the channel. 

It's also not clear where `requestC` getting blocked? Could you help me understand that part better? Also, is this leak something you observed or theoretical?

Reply:
It is a reported bug in pilot ads server. https://github.com/istio/istio/issues/10822

Reply:
It's the same logic in MCP server.